python/cudf/cudf/core/column/decimal.py

```diff
@@ -96,7 +94,12 @@ def _binaryop(self, other: ColumnBinaryOperand, op: str):
                 "__le__",
                 "__ge__",
             }:
-                result = libcudf.binaryop.binaryop(self, other, op, bool)
+                result = libcudf.binaryop.binaryop(lhs, rhs, op, bool)
+            else:
+                raise NotImplementedError(
+                    f"{op} not supported for the following dtypes: "
```

Comment:
This seems to compete with the logic on lines 103-107 which catch libcudf's `RuntimeError` and re-raise a different error about bit widths (too specific?). Shouldn't we be following a single failure path for all these cases, and simply passing through libcudf's `RuntimeError`? I don't like catching that and re-raising `NotImplementedError` here because I don't think we are consistent about our errors across all binary ops and dtypes. I'm content with passing through the `RuntimeError` but not happy with inconsistency if we aren't doing the same for unsupported operators with strings, datetimes, durations, etc.

Reply:
Discussed offline and decided to have one common error message here.